vald/sign.go

```diff
@@ -0,0 +1,113 @@
+package vald
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	ec "github.com/btcsuite/btcd/btcec/v2/ecdsa"
+	"github.com/cosmos/cosmos-sdk/server"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/spf13/cobra"
+
+	"github.com/axelarnetwork/axelar-core/vald/config"
+	"github.com/axelarnetwork/axelar-core/vald/tss"
+	evm "github.com/axelarnetwork/axelar-core/x/evm/types"
+	multisig "github.com/axelarnetwork/axelar-core/x/multisig/exported"
+	"github.com/axelarnetwork/axelar-core/x/tss/tofnd"
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+// GetHealthCheckCommand returns the command to execute a node health check
+func GetSignCommand() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "vald-sign [key-id] [public-key] [hash to sign]",
+		Short: "Sign hash with specified key",
+		Args:  cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			keyID := multisig.KeyID(args[0])
+			if err := keyID.ValidateBasic(); err != nil {
+				return err
+			}
+
+			pubKeyRaw, err := hex.DecodeString(args[1])
+			if err != nil {
+				return err
+			}
+
+			pubKey := multisig.PublicKey(pubKeyRaw)
```

Comment:
Should allow to optionally retrieve the public key from the chain. this will be the default flow since requiring every validator to find their public key for each key id for each chain will be complicated operationally.

Reply:
this will also add the sanity check that the key id is actually registered on chain